### PR TITLE
Anfranci/updatejobexecinput

### PR DESCRIPTION
**Pull request by anandanthony**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by zhoxing-ms** [comment]
@Greedygre Could you please help review this PR?

File: src/containerapp/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 Release History
 ===============
+
+Upcoming
+++++++
+* 'az containerapp job start': update start execution payload format to exlude template property from API version 05-01-2023-preview onwards
```
Comment (Greedygre) [inline src/containerapp/HISTORY.rst:8]:
```suggestion
* 'az containerapp job start': update start execution payload format to exlude template property from API version 2023-05-01-preview onwards
```

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_job_custom_executionstest_e2e_containerappjob.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+            containers:
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_job_custom_executionstest_e2e_containerappjob.yml:2]: please use ` clean_up_test_file(file_name)` remove test yaml file.

File: src/containerapp/azext_containerapp/tests/latest/test_containerappjob_custom_executions.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+from azure.cli.testsdk.scenario_tests import AllowLargeResponse
+from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer, JMESPathCheck, live_only, StorageAccountPreparer)
+from azext_containerapp.tests.latest.common import (write_test_file, clean_up_test_file)
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+from azext_containerapp.tests.latest.common import TEST_LOCATION
+from .utils import create_containerapp_env
+
+class ContainerAppJobsCustomExecutionsTest(ScenarioTest):
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/tests/latest/test_containerappjob_custom_executions.py:18]: Could you please move this test case to exist test file for this command?

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_job_custom_executionstest_e2e_containerappjob.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/tests/latest/test_containerapp_job_custom_executionstest_e2e_containerappjob.yml:1]: please remove this file.

Comment (Greedygre) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_job_custom_executionstest_e2e_containerappjob.yml:1]: Otherwise LGTM

**Review by Greedygre — APPROVED** [review]

File: src/containerapp/azext_containerapp/tests/latest/test_containerapp_job_custom_executionstest_e2e_containerappjob.yml (hunk 4 above)
Comment (anandanthony) [reply src/containerapp/azext_containerapp/tests/latest/test_containerapp_job_custom_executionstest_e2e_containerappjob.yml]: @Greedygre, removed this file

File: src/containerapp/azext_containerapp/custom.py [hunk 5]
```diff
@@ -2151,7 +2151,16 @@ def start_containerappsjob(cmd,
...
                            cpu=None,
                            memory=None,
-                           registry_identity=None):
+                           registry_identity=None,
+                           yaml=None):
+
+    if yaml:
+        if image or container_name or env_vars or\
+            startup_command or args or cpu or memory or\
+                startup_command or args:
```
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/custom.py:2160]: Could we first use some variables with clear meanings to represent this complex expression to increase the readability

File: src/containerapp/azext_containerapp/custom.py [hunk 6]
```diff
@@ -2174,14 +2183,50 @@ def start_containerappsjob(cmd,
...
+    containerappjob_def = None
+
+    # Check if containerapp exists
+    try:
+        containerappjob_def = ContainerAppsJobClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
+    except Exception:
+        pass
+
+    if not containerappjob_def:
+        raise ValidationError("The containerapp job '{}' does not exist".format(name))
```
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/custom.py:2208]:
```suggestion
        raise ResourceNotFoundError("The containerapp job '{}' does not exist".format(name))
```

File: src/containerapp/azext_containerapp/custom.py [hunk 7]
```diff
@@ -2174,14 +2183,50 @@ def start_containerappsjob(cmd,
...
     try:
         return ContainerAppsJobClient.start_job(cmd=cmd, resource_group_name=resource_group_name, name=name, containerapp_job_start_envelope=template_def)
     except CLIError as e:
         handle_raw_exception(e)
 
 
+def start_containerappjob_execution_yaml(cmd, name, resource_group_name, file_name, no_wait=False):
+    yaml_containerappjob_execution = load_yaml_file(file_name)
+    if type(yaml_containerappjob_execution) != dict:  # pylint: disable=unidiomatic-typecheck
+        raise ValidationError('Invalid YAML provided. Please see https://aka.ms/azure-container-apps-yaml for a valid containerapp job execution YAML.')
```
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/custom.py:2197]:
```suggestion
        raise InvalidArgumentValueError('Invalid YAML provided. Please see https://aka.ms/azure-container-apps-yaml for a valid containerapp job execution YAML.')
```

File: src/containerapp/azext_containerapp/custom.py [hunk 8]
```diff
@@ -2174,14 +2183,50 @@ def start_containerappsjob(cmd,
...
+        raise ValidationError("The containerapp job '{}' does not exist".format(name))
+
+    containerappjobexec_def = None
+
+    # Deserialize the yaml into a ContainerApp job execution object. Need this since we're not using SDK
+    try:
+        deserializer = create_deserializer()
+        containerappjobexec_def = deserializer('JobExecutionTemplate', yaml_containerappjob_execution)
+    except DeserializationError as ex:
+        raise ValidationError('Invalid YAML provided. Please see https://aka.ms/azure-container-apps-yaml for a valid containerapp job execution YAML.') from ex
```
Comment (zhoxing-ms) [inline src/containerapp/azext_containerapp/custom.py:2217]:
```suggestion
        raise InvalidArgumentValueError('Invalid YAML provided. Please see https://aka.ms/azure-container-apps-yaml for a valid containerapp job execution YAML.') from ex
```

File: src/containerapp/azext_containerapp/custom.py (hunk 5 above)
Comment (anandanthony) [reply src/containerapp/azext_containerapp/custom.py:2160]: these variables are basically the input that the user provides and are the parameters that are defined for the command. We use the same convention in other methods as well:
For example: https://github.com/Azure/azure-cli-extensions/blob/58560546bb7969c44fb4b46b7ecb7ebff407ad21/src/containerapp/azext_containerapp/custom.py#L425C25-L427C34

**Review by Greedygre — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**